core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -353,17 +352,22 @@ TableScanContext planWith(ExecutorService executor) {
         toSnapshotId,
         executor,
         fromSnapshotInclusive,
-        metricsReporters);
+        metricsReporter);
   }
 
-  Collection<MetricsReporter> metricsReporters() {
-    return metricsReporters;
+  MetricsReporter metricsReporter() {
+    return metricsReporter;
   }
 
   TableScanContext reportWith(MetricsReporter reporter) {
-    ImmutableList.Builder<MetricsReporter> builder = ImmutableList.builder();
-    builder.addAll(metricsReporters);
-    builder.add(reporter);
+    MetricsReporter configured;
+    if (LoggingMetricsReporter.class.equals(metricsReporter.getClass())) {
+      // we're overriding the default LoggingMetricsReporter
```

Comment:
Avoid the use of personal pronouns in comments and documentation. Using "we" or "I" draws unnecessary attention to a subject that doesn't need to be there and makes documentation less direct. Instead, this could be "override the default logging reporter".